privacyidea/lib/monitoringmodules/base.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+#  2018-11-22 Initial create
+#             Cornelius Kölbel <cornelius.koelbel@netknights.it>
+#
+#  License:  AGPLv3
+#  contact:  http://www.privacyidea.org
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+__doc__ = """This is the BaseClass for monitoring statistics data.
+
+The monitoring database is a time series database. It is supposed to store
+tuples of (timestamp, status key, status value)
+
+A monitoring module needs to provide the possibility to write new data,
+return all available keys, return the last data. 
+"""
+import logging
+log = logging.getLogger(__name__)
+from privacyidea.lib.log import log_with
+
+
+class Monitoring(object):
+
+    def __init__(self, config=None):
+        pass
+
+    def get_keys(self):
+        """
+        Return a list of the available statistic keys.
+
+        :return: list of identifiers
+        """
+        return []
+
+    def get_values(self, stats_key, start_timestamp=None, end_timestamp=None, date_strings=False):
```

Comment:
Sorry, this might be nitpicking now, but: I'm not sure about the ``date_strings`` parameter due to two reasons:
1) it changes the return type of the result, which (I think) hinders readability
2) ``get_values`` is implemented by subclasses. But the code handling ``date_strings=True`` would be the same for all subclasses!

So I think it would make more sense to remove the parameter here, and instead have a function ``convert_timestamps_to_stings(values)`` which takes a list of ``(datetime, int)`` tuples and returns a list of ``(string, int)`` tuples. What do you think?

Reply:
Basically you are right.
But: You will run a for loop through a long list of measurement points two times.
And there might be backends, that do not support timestamps anyways but only strings.
Thus different module backends might indeed run a different conversion.

Happy to discuss this.

Reply:
OK. Will do.

Reply:
done in 3679691

Reply:
:+1: